### ISSUE TITLE
org.microg.gms.droidguard should be installed (in /system/priv-app/)

### ISSUE THREAD

**Issue by ArchangeGabriel** [lineageos4microg/l4m_website#14]
It is required that this module is installed as system app (https://github.com/microg/android_packages_apps_RemoteDroidGuard/issues/17, https://github.com/microg/android_packages_apps_GmsCore/issues/482), and @ale5000-git argued that it is not a big deal to have this as default since SafetyNet is initially disabled and people actually have to enable it before DroidGuard could do anything.

Another option could be to provide it as flashable zip and give instructions on how to have it flashed after each OTA update.

**Comment by rugk** [comment]
I'd rather prefer an extra ZIP. I doubt many people SafetyNet.

**Comment by Lanchon** [comment]
a flashable zip could support the addon mechanism to self-reflash after each rom update (as gapps do). however: root, droidguard, xposed, magisk, etc all need flashing and addon support has to be added for each, which is not viable.

a better alternative is needed. the OTA update system should support adding arbitrary zips somehow. it doesn't really need an UI, it'd be enough to have the OTA flash all zips in a special folder in sorted order. (better if the folder tree is searched for zips, in sorted depth-first order.)

the location of the folder itself should be outside of app access for security. '/data/<something>' would be a good choice. which reminds me: i think the OTA might download to the globally accessible sdcard; this is a security risk!

**Comment by ArchangeGabriel** [comment]
Well for instance OmniROM has a folder called “FlashAfterUpdate”. But that is indeed on the `sdcard` storage, as well as the OTA.

**Comment by corna** [comment]
We're still evaluating if DroidGuard should be installed by default (as not everyone needs it, and it downloads and run proprietary software).

> a flashable zip could support the addon mechanism to self-reflash after each rom update (as gapps do). however: root, droidguard, xposed, magisk, etc all need flashing and addon support has to be added for each, which is not viable.

I don't see the issue with the addon mechanism, I think every zip out there already supports it (for example I see that both LineageOS' supersu and magisk support it) and, if not, it's very easy to add it.

**Comment by rugk** [comment]
And a publicly accessible "FlashAfterUpdate" (i.e. on the sd card) is a big security issue considering that any app can them place any zip there, and, as such get root access at the next update. Kinda privilege escalation.

**Comment by ArchangeGabriel** [comment]
@rugk Yes, I get that. Though it’s not any app, it’s any app with storage permission. But I agree this is far from ideal.

@corna As @ale5000-git said, DroidGuard will not download or run anything unless you activate SafetyNet in μG, because it’s off by default. So it is basically harmless on its own.

**Comment by Lanchon** [comment]
> We're still evaluating if DroidGuard should be installed by default (as not everyone needs it, and it downloads and run proprietary software).

i wholeheartedly hope DG is never included by default. SafetyNet would be a total failure if not for google's ability to push it to every handset in the world and enable it against users will. the more devices that don't pass SN, the least SN will be adopted by developers, and the best the community will fare as a whole. i see SN as an anticompetitive practice to lock google competitors (including free software OS providers) out of a larger part of the Android ecosystem. i'm very much against all efforts in the free software community to pass SN, because ultimately it only temporarily lessens the harm of SN and helps defuse the backlash that should be mounting around SN. and all this only ultimately helps google and SN adoption. i firmly believe that some anti-trust organism should force google to, if not remove the product from the market completely, then at least provide a user option to disable checks and report as passing, if the user chooses to do wthout google's "security".

**Comment by corna** [comment]
I agree with @Lanchon, I'm not a fan of DroidGuard either, IMO we should offer at most a ZIP with it.

However at the moment I'm focused on keeping the builds running, so this is not a top-priority for me (but I could review a ZIP made by someone else).

**Comment by Espionage724** [comment]
Does microG need to be installed as a privileged app?

If the ROM is patched to allow location providers outside /system, microG can work fully as a normal app (non-privileged). And from what I've seen, DroidGuard also works (https://github.com/microg/android_packages_apps_GmsCore/issues/482#issuecomment-379459346), but it seems it has to be installed in the same location as GmsCore (both either privileged or non-privileged/normal).

**Comment by corna** [comment]
We use a modified signature spoofing patch, which allows only privileged apps to obtain the FAKE_SIGNATURE permission, so yes, microG (GmsCore) must be installed as privileged app.

**Comment by Lanchon** [comment]
hey @corna, this is OT, but since you mentioned it: could you please, please consider removing the privileged-only restriction on signature spoofing?

there are too many legitimate uses of sigspoof to count. for instance, i publish an apk modding toolchain (https://dexpatcher.github.io/). some apks have an anti-tamper mechanism that check their own signature and can be trivially bypassed with sigspoof. for a more concrete example. i am still stopped from using microg as my daily driver because hangouts doesn't work and i need VoIP access to my google voice number. i could try patching hangouts, except i can't sigspoof it later so i'm sure it will give me trouble.

as a question of principle, users should have as much power as developers on their systems. that's reason enough to have sigspoof open to users.

i understand fears of misuse by apps, but the privileged-only solution sucks. a global developer setting could be added to enable sigspoof for non-privileged apps instead. code for such a setting can be found on some of marvin patches or [here](https://github.com/Lanchon/haystack/tree/master/patches-src/sigspoof-ui-global-7.0-7.1). please note that per-app runtime permission requesting dialog should not be shown at all if the global setting is disabled.

another solution would be a hidden setting without a UI. for example, the user creating a special file in say /data/somewhere/special_file would enable global sigspoof. whatever works, just please do something.

thank you!!!

**Comment by rugk** [comment]
So where is this issue?

Also BTW, the privileged-only restriction on signature spoofing is a **security feature**. You can't just remove that, because then the claim that LineageOS devs have that microG introduces security issues is true. Because if it is unlimited, every app can spoof any app, including permissions etc. (At least taht's how I understand it.)
You could, however, theoretically limit signature spoofing to all microG apps, IMHO. I don't know any other app, which would have a reason to use it, anyway.

**Comment by corna** [comment]
@rugk Let's not pollute this issue, we can discuss this in a new one. In short, I won't open the signature spoofing to non-privileged apps, but I have a solution that might get everyone happy.

**Comment by Lanchon** [comment]
privileged sigspoof discussion moved [HERE](https://github.com/lineageos4microg/docker-lineage-cicd/issues/11).

**Comment by petefoth** [comment]
Moving this issue to the [docker-lineage-cicd](https://github.com/lineageos4microg/docker-lineage-cicd/issues) issue tracker

This issue tracker is only for issues with our website https://lineage.microg.org/

Please go to the web site, and scroll down to the section 'Where do I report a bug?'. Check out the diagram, and report the issue in the correct place.

Please also see the [Troubleshooting and support](https://github.com/lineageos4microg/docker-lineage-cicd#troubleshooting-and-support) and [Issue Reporting & Tracking](https://github.com/lineageos4microg/docker-lineage-cicd#issue-reporting--tracking) section in the README.md

Thanks
